pkgs/applications/version-management/git-and-tools/ghorg/default.nix

```
@@ -0,0 +1,27 @@
+{ stdenv, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "ghorg";
+  version = "1.4.0";
+
+  src = fetchFromGitHub {
+    owner = "gabrie30";
+    repo = "ghorg";
+    rev = "${version}";
+    sha256 = "0diwndkckv6fga45j9zngizycn5m71r67cziv0zrx6c66ssbj49w";
+  };
+
```

```suggestion
vendorSha256 = stdenv.lib.fakeSha256;
```

(And edit the value after.)

---

Thanks for pointing out the `vendorSha256` bit, @AndersonTorres. I wasn't even aware of this.

It seems like project authors can choose a [vendoring approach](https://golang.org/ref/mod#vendoring) that works for them. Some go packages in nixpkgs also don't list `vendorSha256` and based on pkgs/development/go-modules/generic/default.nix vendorSha256 is allowed to be null (default) if there already is a vendor directory within the source. Do I have to set `deleteVendor = true` in order to deal with the existence of a vendor directory for this particular project and retrieve the vendored packages during build?

On my setup, I managed to build the ghorg package, and I'm still very much of a noob at times, so some guidance towards a solution would be greatly appreciated.

---

For now just update vendorSha256.

---

When I add `vendorSha256`, I get:

```
building '/nix/store/6p534hgsv7095nbl2fid89k0p24kyzcs-ghorg-1.4.0-go-modules.drv'...
unpacking sources
unpacking source archive /nix/store/1jmi9xxkfh7xm2dxbngj28vbkjawh0s5-source
source root is source
patching sources
configuring
building
vendor folder exists, please set 'vendorSha256 = null;' in your expression
builder for '/nix/store/6p534hgsv7095nbl2fid89k0p24kyzcs-ghorg-1.4.0-go-modules.drv' failed with exit code 10
cannot build derivation '/nix/store/4bhw0ibmnm8lc8isdvihajv3na1va68d-ghorg-1.4.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/4bhw0ibmnm8lc8isdvihajv3na1va68d-ghorg-1.4.0.drv' failed
```

so I'm guessing I have to file the package
1. with `vendorSha256` undefined OR
2. with `deleteVendor = true;` and `vendorSha256` defined.

---

(Sorry, I am still learning about specific packaging...)

---

So... `vendorSha256 = null` and `doCheck = false` builds :+1:

I did however find a way to run the tests by setting `$HOME` to the `/build` directory. Is this a bad practice? I figured that this may pass the smell test as the change to /build is deterministic since it's just a copy from a file already in the source to a "legal" path.

I thought this would be better in case the project adds some decent tests down the line :wink:.

If you decide that this reeks, I'll just push the `vendorSha256 = null` and `doCheck = false` bit such that we can move this along. Thanks for your input, we're almost there. :smile:

---

For now, I would push `vendorSha256 = null` and `doCheck = false`.

In this particular case I think the upstream developers need to modify their product. It is not a good practice to use $HOME as placeholder for tests. It could be even flagged as a security flaw.